src/main/db_main.cpp

```diff
@@ -0,0 +1,52 @@
+#include "main/db_main.h"
+#include <memory>
+
+namespace terrier {
+
+int DBMain::Init(int argc, char **argv) {
```

Comment:
I am not convinced these are the right arguments. At the very least, GFlags also need to process these raw command args. I think a better idea would be to separate out the interpretation of these command line args and the initialization of `SettingsManager` into their own function or module.

Reply:
@tli2 Is correct here. `DBMain` should **not** have to process commandline arguments. If we did it this way, then every test case would need to hardcode the arguments. This will make our code brittle.

Reply:
Then we need to bring up `SettingsManager` before bringing up `DBMain`?

Reply:
We have a map passed to the DB Main object now.